chia/server/server.py

```diff
@@ -615,7 +616,14 @@ async def send_to_all(self, messages: List[Message], node_type: NodeType):
                     if sent_message_response_ok(ProtocolMessageTypes(message.type), None):
                         await connection.send_message(message)
                     else:
-                        self.log.error(f"send_to_all not sending message expecting a response: {message.type}")
+                        # Internal peer protocol logic error
+                        # If we raise ProtocolError here exiting the loop will affect other peers
+                        self.log.error(f"send_to_all called with message expecting a response: {message.type}")
+                        await connection.close(
```

Comment:
These two calls to `connection.close` with `WSCloseCode.INTERNAL_ERROR` are because we do not expect to call `send_message` with peer messages that expect a reply.

See `chia/server/ws_connection.py` send_message

Reply:
if we accidentally do this, and it's triggered by a test, we won't necessarily fail the test, because tests are allowed to print errors to the log

Reply:
@arvidn @almogdepaz What's a good solution here?

Should we save the fact that we had an error, and throw at the end of the loop?
Assert during tests?

Reply:
Ah, you know what, I can raise here, because the check is on the message type - it would affect all clients.

Reply:
I think `raise` would be reasonable